retrieval/client.go

```diff
@@ -98,12 +98,6 @@ type clientStream struct {
 // > DealProposal(...)
 // < ...
 func (c *Client) RetrieveUnixfs(ctx context.Context, root cid.Cid, size uint64, total types.BigInt, miner peer.ID, client, minerAddr address.Address, out io.Writer) error {
-	s, err := c.h.NewStream(ctx, miner, ProtocolID)
-	if err != nil {
-		return xerrors.Errorf("failed to open stream to miner for retrieval query: %w", err)
-	}
-	defer s.Close()
-
```

Comment:
We should make sure that we can actually talk to the peer - simplest way would be to use libp2p ping. For one example, see
https://github.com/ipfs/go-ipfs/blob/master/core/commands/ping.go#L106

Reply:
Please check this out